### ISSUE TITLE
Error: ECONNRESET on Email receive

### ISSUE THREAD

**Issue by Z3roCoder** [djfarrelly/MailDev#72]
Hello.
node -v: 0.12.4
npm version: 2.11.2

When running maildev and upon receiving an Email, maildev instantly crashes for me with the following log:
```
MailDev app running at 0.0.0.0:1080
MailDev SMTP Server running at 0.0.0.0:1025
createCredentials() is deprecated, use tls.createSecureContext instead
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: read ECONNRESET
    at exports._errnoException (util.js:746:11)
    at TCP.onread (net.js:559:26)
```
Any idea what it could be?

Steps to reproduce:

PS> npm install -g maildev
PS> maildev
Receive email from anywhere

**Comment by Seidr** [comment]
I ran in to this yesterday while making some changes to MailDev locally. I believe this is due to the mail client that is sending the email hanging up ungraciously. It also has something to do with TLS being enabled. You can replicate this with the following command.

```echo "This is the message body and contains the message" | mailx -v -r "test@test.com" -s "This is the subject" -S smtp="localhost:1025" -S smtp-use-starttls -S ssl-verify=ignore test@test.com```

The same command can be made to execute successfully by disabling TLS.

```echo "This is the message body and contains the message" | mailx -v -r "test@test.com" -s "This is the subject" -S smtp="localhost:1025" test@test.com```

As for the why..I'm not sure..but I'll have a dig during my break and see if I can come up with a better explanation.

**Comment by Z3roCoder** [comment]
Hmm, you are right. Turning off TLS from my testing gitlab server enabled maildev to continue working and not spit out such messages. That's good to know, thank you!

**Comment by Seidr** [comment]
Ok, it looks like it IS due to the client hanging up prematurely. I can still use TLS, while preventing MailDev from crashing by commenting out the following line with the 'this._destroy()' call in the file 'node_modules/simplesmtp/node_modules/rai/lib/rai.js'

```/**
 * <p>Called when an error has appeared. Emits <code>'error'</code> with
 * the error object as a parameter.</p>
 *
 * @event
 * @param {Object} err Error object
 */
RAISocket.prototype._onError = function(err){
    this.emit("error", err);
//    this._destroy();
};```

While this is not a great solution, as simplesmtp is going to be (I think) replaced in the future, I'm not sure if it's worth doing a pull request for this.

**Comment by djfarrelly** [comment]
Great digging in on this one guys! This looks related to #68. simplesmtp doesn't play nicely with Node 0.12.x it seems. It may finally be time to take care of replacing simplesmtp (#30). I also would love to write a failing test for this one to run against.

**Comment by djfarrelly** [comment]
Fixed with version 0.11.0
